### PR TITLE
Add Piotr's suggestions to Daemon page

### PR DESCRIPTION
**Pull request by nathan-contino**
Polish Daemon page based on Piotr's suggestions.

### Context
Constant vigilance is our best tool when it comes to the daemon.

Rendered docs pages:
[The Gradle Daemon](https://builds.gradle.org/repository/download/Gradle_Release_Check_BuildDistributions/56245928:id/distributions/gradle-7.6-docs.zip!/gradle-7.6-20220919155611%2B0000/docs/userguide/gradle_daemon.html)
[File System Watching](https://builds.gradle.org/repository/download/Gradle_Release_Check_BuildDistributions/56245928:id/distributions/gradle-7.6-docs.zip!/gradle-7.6-20220919155611%2B0000/docs/userguide/file_system_watching.html#sec:daemon_watch_fs)

### Contributor Checklist
- [X] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [X] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [X] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [X] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [X] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [X] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [X] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [X] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [X] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/running-builds/gradle_daemon.adoc [hunk 1]
```diff
@@ -29,6 +29,36 @@ The Daemon reduces build times by:
 * benefiting from continuous runtime optimization in the JVM
 * watching the file system to calculate exactly what needs to be rebuilt before you run a build
```
Comment (nathan-contino) [inline subprojects/docs/src/docs/userguide/running-builds/gradle_daemon.adoc:30]:
```suggestion
* <<file_system_watching.adoc#sec:daemon_watch_fs,watching the file system>> to calculate exactly what needs to be rebuilt before you run a build
```

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Review by hythloda — APPROVED** [review]

**Comment by nathan-contino** [comment]
@bot-gradle test and merge

**Comment by nathan-contino** [comment]
@bot-gradle test and merge

**Pull request merged**